pkg/cmd/codespace/ports.go

```diff
@@ -249,14 +275,46 @@ func (a *App) UpdatePortVisibility(ctx context.Context, codespaceName string, ar
 	}
 	defer safeClose(session, &err)
 
+	g, ctx := errgroup.WithContext(ctx)
+
 	// TODO: check if port visibility can be updated in parallel instead of sequentially
 	for _, port := range ports {
 		a.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating port %d visibility to: %s", port.number, port.visibility))
-		err := session.UpdateSharedServerPrivacy(ctx, port.number, port.visibility)
-		a.StopProgressIndicator()
-		if err != nil {
-			return fmt.Errorf("error update port to public: %w", err)
+
+		// wait for success or failure
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+
+		g.Go(func() error {
+			updateNotif, err := session.WaitForPortNotification(ctx, port.number, liveshare.PortChangeKindUpdate)
+			if err != nil {
+				return fmt.Errorf("error waiting for port %d to update: %w", port.number, err)
+
+			}
+			if !updateNotif.Success {
+				if updateNotif.StatusCode == http.StatusForbidden {
+					return newErrUpdatingPortVisibility(port.number, port.visibility, errUpdatePortVisibilityForbidden)
+				}
+				return newErrUpdatingPortVisibility(port.number, port.visibility, errors.New(updateNotif.ErrorDetail))
+
+			}
+			return nil // success
+		})
+
+		g.Go(func() error {
+			err := session.UpdateSharedServerPrivacy(ctx, port.number, port.visibility)
+			if err != nil {
+				return fmt.Errorf("error updating port %d to %s: %w", port.number, port.visibility, err)
+			}
+			return nil
+		})
+
+		// wait for success or failure
+		if err := g.Wait(); err != nil {
+			return err
```

Comment:
I think you want to stop the progress indicator here too:

```
err := g.Wait()
a.StopProgressIndicator()
if err != nil {
	return err
}
```

Alternatively, assuming you don't need to call stop when you go round the loop again, you could hoist the stop call above the loop and use 'defer' to ensure that it always happens on return.

Reply:
good catch, fixed in latest commit